pinot-common/src/main/java/org/apache/pinot/common/utils/URIUtils.java

```diff
@@ -70,6 +70,14 @@ public static String getPath(String basePath, String... parts) {
     return stringJoiner.toString();
   }
 
+  /**
+   * Returns the last part for the given path split by the file separator.
+   * If the file separator is not found, returns the whole path as the last part.
+   */
+  public static String getLastPart(String path) {
```

Comment:
If used at other places, the URI might have query parameters. Should those be removed? Or add a test case for "?..="

Reply:
Good catch! Added a few more unit tests.